app/models/payment.rb

```diff
@@ -105,40 +109,46 @@ def initialize(bt_result, page_id, member_id, existing_customer, save_customer=t
       @bt_result = bt_result
       @page_id = page_id
       @member_id = member_id
-      @existing_customer = existing_customer
       @save_customer = save_customer
+      @existing_customer = existing_customer
     end
 
     def build
       return unless transaction.present?
-      ::Payment::BraintreeTransaction.create(transaction_attrs)
+      # a Payment::BraintreeCustomer gets created for both successful and failed transactions. The customer_id will be nil,
+      # though, because trasnaction.customer_details.id is nil for failed transaction.
+      @customer = @existing_customer || Payment::BraintreeCustomer.find_or_create_by!(
+          member_id: @member_id,
+          # TODO: This is a problem for webhook notifications. At least in the example notification in the specs,
+          # there are no customer details, so a new customer will always be created with an existing member_id
+          # and a nil customer_id. This won't fly.
```

Comment:
Because we want to be able to associate failed transactions with the people who made them. They're associated with members through the Payment::BraintreeCustomer class. This won't create a problem for webhooks because the locally stored customer will be updated with a valid `customer_id` from BrainTree when they make that successful subscription.